secboot/bootstrap_key.go

```diff
@@ -0,0 +1,104 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2024 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package secboot
+
+import (
+	"fmt"
+
+	"github.com/snapcore/snapd/osutil"
+)
+
+type DiskUnlockKey = []byte
+
+type KeyDataWriter interface {
+	// TODO: this will typically have a function that takes a key
+	// data as input
+}
+
+// BootstrapKey is an abstraction that contains a key container with
+// its bootstrap key and allows adding new keys to the container, then
+// later clean the bootstrap key once all keys were enrolled.
+type BootstrapKey interface {
```

Comment:
should this maybe include 'container' in the name?

Reply:
What about `BootstrapContainer`. And then we document it as "A BootstrapContainer is an encrypted container (LUKS2 or other) that has a bootstrap key able to add other keys."

Reply:
then maybe BootstrapKeysContainer maybe?

Reply:
my preference are either BootstrappedContainerWithKey or BootstrappedContainer

Reply:
I have renamed it to BootstrappedContainer